qabelbox/src/main/java/de/qabel/desktop/repository/sqlite/SqliteContactRepository.kt

```diff
@@ -0,0 +1,344 @@
+                    }
+                    return hydrator.hydrateOne(results)
+                })
+            }
+        } catch (e: SQLException) {
+            throw EntityNotFoundException("exception while searching contact: " + e.message, e)
+        }
+
+    }
+
+    @Throws(EntityNotFoundException::class)
+    override fun findByKeyId(keyId: String): Contact {
+        try {
+            database.prepare(
+                    "SELECT " + hydrator.getFields("c").joinToString() + " FROM " +
+                            ContactDB.TABLE_NAME + " " + ContactDB.TABLE_ALIAS + " " +
+                            "WHERE c.publicKey = ? " +
+                            "LIMIT 1")
+                    .use { statement ->
+                        statement.setString(1, keyId)
+                        statement.executeQuery().use({ results ->
+                            if (!results.next()) {
+                                throw EntityNotFoundException(
+                                        "no contact found for key '$keyId'")
+                            }
+                            return hydrator.hydrateOne(results)
+                        })
+                    }
+        } catch (e: SQLException) {
+            throw EntityNotFoundException("exception while searching contact: " + e.message, e)
+        }
+
+    }
+
+    @Throws(PersistenceException::class, EntityNotFoundException::class)
+    override fun findContactWithIdentities(key: String): Pair<Contact, List<Identity>> {
+        val contact = findByKeyId(key)
+
+        val identities = identityRepository.findAll()
+        val associatedIdentities = findContactIdentityKeys(listOf(contact.id))
+
+        val contactIdentities: List<Identity>
+        if (associatedIdentities.contains(contact.id)) {
+            contactIdentities = associatedIdentities[contact.id]!!
+                    .map { identityKey -> identities.getByKeyIdentifier(identityKey) };
+        } else {
+            contactIdentities = emptyList<Identity>()
+        }
+        return Pair(contact, contactIdentities)
+    }
+
+
+    @Throws(PersistenceException::class)
+    private fun find(searchString: String?): Collection<Contact> {
+        val hydrator = SimpleContactHydrator(entityManager)
+        val queryBuilder = StringBuilder(
+                "SELECT " + StringUtils.join(",", hydrator.getFields("c")) + " " +
+                        "FROM contact c ")
+
+        val filterResults = searchString != null && !searchString.trim().isEmpty()
+        if (filterResults) {
+            queryBuilder.append("WHERE (lower(c.alias) LIKE ? OR c.phone LIKE ? OR c.email LIKE ?)")
+        }
+
+        try {
+            database.prepare(queryBuilder.toString()).use { statement ->
+                var paramIndex = 1
+                if (filterResults) {
+                    val lowerWildSearchString = searchString!!.toLowerCase() + "%"
```

Comment:
I think that having this argument nullable is a mistake.

```
private fun find(searchString: String = "") {
    ...
    val trimmedSearch = searchString.trim().toLowerCasee()
    if (!trimmedSearch.isEmpty()) { ...}
    ...
    if (!trimmedSearch.isEmpty()) {
        val wildCardSearch = trimmedSearch + "%"
        ....
```

Splitting those two functions in one (because searchString atcually is a flag-parameter) and putting the burden of the nullable search someplace else wouldn't be a bad idea either.